### PR TITLE
test: adding slashable shares in queue

### PR DESCRIPTION
**Pull request by eigenmikem**
**Motivation:**

Slashable shares in queue aren't currently checked by any invariants.

**Modifications:**

- added assertion that slashable shares in queue decrease on slash if a withdrawal is queued
- added assertion that slashable shares in queue increase on queued withdrawal
- changed IntegrationDeployer to initialize starting block as nonzero to address edge cases

**Result:**

Invariant coverage for slashable shares in queue state

### CODE REVIEW

File: src/test/integration/IntegrationChecks.t.sol [hunk 1]
```diff
@@ -997,6 +999,35 @@ contract IntegrationCheckUtils is IntegrationBase {
         // assert_Snap_Unchanged_AllocatedStrats(operator, operatorSet, "should not have updated allocated strategies");
     }
 
+    function check_Base_Slashing_State(
+        User operator,
+        AllocateParams memory allocateParams,
+        SlashingParams memory slashParams,
+        Withdrawal[] memory withdrawals
+    ) internal {
+        OperatorSet memory operatorSet = allocateParams.operatorSet;
```
Comment (ypatil12) [inline src/test/integration/IntegrationChecks.t.sol:1008]: Nit: this can just call the previous function, and then we add the withdrawal specific check

**Review by ypatil12 — COMMENTED** [review]
Looks good, one minor comment

File: src/test/integration/IntegrationChecks.t.sol [hunk 2]
```diff
@@ -997,6 +999,16 @@ contract IntegrationCheckUtils is IntegrationBase {
         // assert_Snap_Unchanged_AllocatedStrats(operator, operatorSet, "should not have updated allocated strategies");
```
Comment (ypatil12) [inline src/test/integration/IntegrationChecks.t.sol:999]: How can we assert that the slashable shares in queue decreases after a withdrawal is completable ?

Comment (ypatil12) [reply src/test/integration/IntegrationChecks.t.sol:999]: Down to add it to roll blocks

Comment (ypatil12) [reply src/test/integration/IntegrationChecks.t.sol:999]: Nvm, discussed offline, can put it in `check_Complete_Withdrawal_AsX_State`. We call this function right when withdrawal is completable so this is fine

**Review by ypatil12 — APPROVED** [review]
LGTM, we'll add the invariant for slashable shares in queue decreasing after a withdrawal is completable separately (if possible)
